trio-stubs/__init__.pyi

```diff
@@ -84,7 +85,12 @@ class Cancelled(BaseException, _NotConstructible, metaclass=ABCMeta):
 # _core._multierror
 class MultiError(BaseException):
     exceptions: List[BaseException]
-    def __init__(self, exceptions: Sequence[BaseException]): ...
+    def __init__(
+        self, exceptions: Sequence[BaseException], *, _collapse: bool = ...
+    ): ...
+    def __new__(
+        cls: T, exceptions: Sequence[BaseException], *, _collapse: bool = ...
+    ) -> T: ...
```

Comment:
Probably better to _not_ expose this param in stubs, since it's got a leading underscore and use of `MultiError` is deprecated anyway!

Reply:
I meant "have a stub for MultiError, which omits the `_collapse` parameter", I think we agree on the first part but I miscommunicated about the second?

Reply:
Ah no that's how I interpreted you :sweat_smile: So the miscommunication must be on my part! :D

I think adding `trio.MultiError.__init__` and `trio.MultiError.__new__` to `allowlist.txt` isn't great. Another reason for keeping the stubs technically correct is to ease a transition away from having separate stub files at all, and if exposing `_collapse` is bad for some reason that should be addressed in `trio` - rather than hidden by having an "incorrect" type for the function in the stub.